source/slang/slang-emit-glsl.cpp

```diff
@@ -1404,21 +1404,48 @@ bool GLSLSourceEmitter::tryEmitInstExprImpl(IRInst* inst, const EmitOpInfo& inOu
         case kIROp_BitCast:
         {
             auto toType = extractBaseType(inst->getDataType());
+            auto fromType = extractBaseType(inst->getOperand(0)->getDataType());
```

Comment:
Maybe the original form of the logic got lost, but at some point I had intended for this conversion operations to implement the conversion in two steps to keep things simple:

1. Convert `fromType` to the same-size `uint` type, which we can consider the "intermediate" type.
2. Convert the intermediate type to `toType`, with the knowledge that the intermediate is always a `uint` type of matching size

The alternative would seem to be having nested `switch` statements in the long run, which is obviously not ideal.